src/site/cpp-objects.rst

```diff
@@ -0,0 +1,171 @@
+**************************
+C++ Lessons #5 --- Objects
+**************************
+
+.. warning::
+
+    * For these, I will only go over the C++ details at a high level
+    * Getting into the nitty gritty of C++ is not the purpose of this aside
+    * Some of the concepts require knowledge of more advanced ideas that have not been covered yet
+        * If this happens, do not panic
+        * it probably makes more sense to revisit this later in the course
+    * For simplicity, some of the examples will use less than ideal implementations
+
+
+* The same example described in :doc:`Topic3 </topic3.rst>` is used here
+    * A collection of ``Friend`` objects stored in an ``ContactList`` obejct
+
+
+Friend
+======
+
+* We will create a ``Friend`` class to keep track of a few details
+    * First Name
+    * Last Name
+    * Email
+
+* We will also want
+    * Constructor
+    * Accessors
+
+* That's about it
+
+
+Header
+------
+
+* The first thing we want is a *header* file
+* Call this ``Friend.h``
+* This is where we typically declare all the member fields and methods the class will have
+* This file will also have all our ``#include``s too
+* Below is a complete ``Friend.h``
+
+
+.. code-block:: cpp
+    :linenos:
+
+    // Friend.h
+    #include <string>
+
+    class Friend {
+        private:
+            std::string firstName_;
+            std::string lastName_;
+            std::string email_;
+
+    public:
+        Friend();
+        Friend(std::string, std::string, std::string);
+
+        const std::string& firstName() const    {return firstName_;}
+        const std::string& lastName() const     {return lastName_;}
+        const std::string& email() const        {return email_;}
+              std::string& email()              {return email_;}
+    };
+
+
+* The last four lines are a way we can have accessors for a private fields
+* The first thing to note is that the whole method is actually written here in the header since it is only one line long
+* The first two return ``const std::string&`` --- a constant reference to a string
+    * These values can be accessed, but not changed
+    * You can access the like this ``someFriend.firstName()``
+
+* ``email()`` on the other hand has another method that does not return a ``const``, which means that this value may be changed
+    * You can modify the email like this ``someFriend.email() = "new email address"``
+
+
+Class
+-----
+
+* The class file is where we typically define and write the actual methods
```

Comment:
`.cpp` files are usually refered to as implementation files. You can add whatever to them, including functions that aren't part of a class, so the name _class_ is more java-esque again.